src/shared/Core/Trace2.cs

```diff
@@ -219,6 +215,7 @@ private void TryParseSettings(TextWriter error, IFileSystem fileSystem)
         {
             Event = Trace2Event.Version,
             Sid = _sid,
+            Thread = Thread.CurrentThread.Name,
```

Comment:
> Other thread names are of the form "th%d:%s" and include a unique number and the name of the thread-proc.

Should we not be trying to fit to the same format? Is it 'required'? @jeffhostetler

```csharp
int tid = Thread.CurrentThread.ManagedThreadId;
string tname = Thread.CurrentThread.Name;
$"th{tid}:{tname}"
```

Reply:
@jeffhostetler - this is a good question. Since GCM is going to be a child process of Git should we follow this naming convention for GCM-related threads rather than using `main`?

Reply:
GCM is it's own process, so it should have a "main" thread.  The main thread and any children, threads, and regions that GCM creates will form their own distributed trace graph.  It just happens that by SID magic, I can combine the graphs of the invoking Git command (like "git push") and the child Git commands that you spawn (like "git config") in to a single very large graph.

Within GCM, your threads can be named in anyway that is helpful to you.  I was limited because of C and coding conventions in how I could create thread names.  OTOH, C# has some thread name magic, but it would be better to
avoid some of the generic or lambda or pool thread name stuff.  The important part is that all of the Trace2 events
for an activity send the same thread name -- and not just the pool thread name.  I need to be able to tie the
events together on an activity as one that is running in parallel with other such activities.

The exact format "th%d:%s" doesn't so much matter.  However, that format name will show up on the graph.

Reply:
Sounds good! Thanks for the thorough explanation, Jeff.